docs/js/pricking.js

```diff
@@ -0,0 +1,86 @@
+    var shapeId = shape.attributes["id"].value
+    var action = document.querySelector('input[name = "stitch"]:checked').value
+    if (action == "remove") {
+        // for now show the event is applied, custom link shows the calculated value
+        shape.getElementsByTagName("title")[0].innerHTML = `${shapeId} : deleted from custom link`
+        shape.style = `fill:${color};opacity:0.0`
+        var currentLinks = link.href.replace(/.*=/,"")
+        var newlinks = TopoLink.removeStitch(shapeId,currentLinks)
+        link.href = "?topo=" + newlinks
+        showDiagram(newlinks)
+    } else {
+        var color = document.getElementById("color").value
+        var stitch = document.getElementById("stitch").value
+        var cx = shape.getAttribute("cx") * 1
+        var cy = shape.getAttribute("cy") * 1
+        switch(document.getElementById("shape").value) {
+          case "square":
+            shape.outerHTML = `<rect x="${cx - 8}" y="${cy - 8}" width="16" height="16" style="fill:${color};opacity:0.65" id="${shapeId}" ><title>${shapeId} ${stitch}</title></rect>`
+            break;
+          case "pyramid":
+            var points = `${cx + 11},${cy + 6} ${cx - 11},${cy + 6} ${cx},${cy - 11}`
+            shape.outerHTML = `<polygon points="${points}" style="fill:${color};opacity:0.65" id="${shapeId}" ><title>${shapeId} ${stitch}</title></rect>`
+            break;
+          case "triangle":
+            var points = `${cx + 11},${cy - 6} ${cx - 11},${cy - 6} ${cx},${cy + 11}`
+            shape.outerHTML = `<polygon points="${points}" style="fill:${color};opacity:0.65" id="${shapeId}" ><title>${shapeId} ${stitch}</title></rect>`
+          case "star":
+            var points = `${cx + 17.46621},${cy + 19.210733} C ${cx + 16.358277},${cy + 19.992804} ${cx + 13.888562},${cy + 16.974202} ${cx + 12.532536},${cy + 16.955684} ${cx + 11.17651},${cy + 16.937165} ${cx + 8.6252843},${cy + 19.887199} ${cx + 7.5391212},${cy + 19.075165} ${cx + 6.452958},${cy + 18.263131} ${cx + 8.5606347},${cy + 14.981493} ${cx + 8.1592117},${cy + 13.686113} ${cx + 7.7577888},${cy + 12.390733} ${cx + 4.1637682},${cy + 10.875984} ${cx + 4.6004154},${cy + 9.5920494} ${cx + 5.0370626},${cy + 8.3081146} ${cx + 8.8093936},${cy + 9.2985524} ${cx + 9.9173265},${cy + 8.5164821} ${cx + 11.025259},${cy + 7.7344118} ${cx + 11.355258},${cy + 3.8482119} ${cx + 12.711284},${cy + 3.8667304} C ${cx + 1.356026},${cy + 0.018518} ${cx + 1.579778},${cy + 3.9122805} ${cx + 2.665941},${cy + 4.7243144} ${cx + 1.086163},${cy + 0.8120338} ${cx + 4.884134},${cy - 0.075021} ${cx + 5.285557},${cy + 1.2203591} ${cx + 0.401423},${cy + 1.2953801} ${cx - 3.232621},${cy + 2.7114191} ${cx - 3.669269},${cy + 3.9953541} ${cx - 0.436647},${cy + 1.283935} ${cx + 1.58063},${cy + 4.621905} ${cx + 0.472697},${cy + 5.403975}`
+            shape.outerHTML = `<path d="M ${points} Z" style="fill:${color};opacity:0.65" id="${shapeId}" />`
```

Comment:
@d-bl/gf 
Who helps trouble shooting this star shape?
![image](https://user-images.githubusercontent.com/10553630/105635130-aea7c000-5e61-11eb-86aa-a4cbfd91252c.png)

You can try it on https://jo-pol.github.io/GroundForge/pricking

The star gets messed up:
![image](https://user-images.githubusercontent.com/10553630/105635086-77391380-5e61-11eb-9812-b48c97ae772f.png)

Another bug (beyound the question above) emerges when changing the same stitch a second time.

Reply:
explanation of the point string: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d

another view on the same shape:
![image](https://user-images.githubusercontent.com/10553630/105756830-6f569d80-5f4d-11eb-89d8-e8b605d7b2e2.png)

Reply:
and what is intended as a crescent is rendered as a pentagon

Reply:
and what is intended as a crescent is rendered as a pentagon

Reply:
The problem is not caused by changing a rendered diagram. We get the same shapes when changing the complete diagram:

https://github.com/d-bl/GroundForge/blob/e81f2621ee1e872ce7548f94b08036b6179dc70f/src/main/scala/dibl/fte/SvgPricking.scala#L44-L47

The crescent that turns into a pentagon:
```scala
    val dots = nodes.map{ case (id,(x,y)) =>
      val cx = scale * x + offsetX - 12
      val cy = scale * y + offsetY - 12
      val points = s"${cx + 20.4428},${cy + 12.109177}  A ${cx + 8},${cy + 8}  0 0 1 ${cx + 15.498201},${cy + 19.502723}  ${cx + 8},${cy + 8}  0 0 1 ${cx + 6.7766691},${cy + 17.756739}  ${cx + 8},${cy + 8}  0 0 1 ${cx + 5.0593222},${cy + 9.0295237}  ${cx + 8},${cy + 8}  0 0 1 ${cx + 12.469058},${cy + 4.1092197}"
      val path = s"""<path d="M $points Z" style="fill:rgb(225,0,0);opacity:0.65" id="$id"><title>$id</title></path>"""
      s"""<a href="#" onclick="clickedDot(this)">$path</a>"""
    }
```

Reply:
@veronika The promised SVG for the first screenshot in this sequence of messages, with the green cresent and red star.
[shapes.svg.zip](https://github.com/d-bl/GroundForge/files/5880864/shapes.svg.zip)

In the end I intend to implement it somehow in scala. For now JavaScript provides a quick trial and error test cycle.